signal/signal.go

```diff
@@ -45,18 +56,19 @@ func Intercept() error {
 		syscall.SIGTERM,
 		syscall.SIGQUIT,
 	}
-	signal.Notify(interruptChannel, signalsToCatch...)
-	go mainInterruptHandler()
+	signal.Notify(channels.interruptChannel, signalsToCatch...)
+	go channels.mainInterruptHandler()
 
-	return nil
+	return channels, nil
 }
 
 // mainInterruptHandler listens for SIGINT (Ctrl+C) signals on the
 // interruptChannel and shutdown requests on the shutdownRequestChannel, and
 // invokes the registered interruptCallbacks accordingly. It also listens for
 // callback registration.
 // It must be run as a goroutine.
-func mainInterruptHandler() {
+func (c *Interceptor) mainInterruptHandler() {
+	defer atomic.StoreInt32(&started, 0)
```

Comment:
to we still need this global var?

Reply:
I am not sure, right now `signal.Intercept` checks the value of it, as to prevent another signal interception.

https://github.com/lightningnetwork/lnd/blob/68c07352eb3279b9b91292509d369c6f43587b7a/signal/signal.go#L41-L44

Reply:
I think this can remain as is, can't see a case for multiple interceptors in the same process? And we would need to change some of the other functions, so perhaps in a followup if we really feel the need to kill all the globals.

Reply:
Yes, I suggest we leave the started global as is, removing or changing it seems to open up more questions than would close at the moment.
It would be the last thing to fix to make the signal package completely "non-global".